pkg/preparation/shards/model/shard.go

```diff
@@ -0,0 +1,114 @@
+package model
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/ipfs/go-cid"
+	"github.com/storacha/guppy/pkg/preparation/types"
+	"github.com/storacha/guppy/pkg/preparation/types/id"
+)
+
+// ShardState represents the state of a shard.
+type ShardState string
+
+const (
+	// ShardStateOpen indicates that a shard is still accepting new nodes.
+	ShardStateOpen ShardState = "open"
+	// ShardStateClosed indicates that a shard is no longer accepting nodes, but
+	// is not yet added to the space.
+	ShardStateClosed ShardState = "closed"
+	// ShardStateAdded indicates that a shard has been added to the space.
+	ShardStateAdded ShardState = "added"
+)
+
+func validShardState(state ShardState) bool {
+	switch state {
+	case ShardStateOpen, ShardStateClosed, ShardStateAdded:
+		return true
+	default:
+		return false
+	}
+}
+
+type Shard struct {
+	id       id.ShardID
+	uploadID id.UploadID
+	cid      *cid.Cid
```

Comment:
why *cid.Cid here and cid.Undef on dagScan?

Reply:
Uh, because I changed the other one and forgot about this one? 😅 I'll make this one match.